laurent22/joplin
# Desktop: Resolves #8931: Improve support for plugins in the Rich Text Editor (implement `webviewApi.postMesage`)

# Summary

This pull request makes it possible for a plugin's content scripts to communicate with its background script, without using internal Joplin APIs (e.g. through `parent.require`). Specifically, it
1. Allows renderer plugins to access `webviewApi` when running in the rich text editor. This is the same API as is used in the markdown preview pane.
2. Updates `tests/support/plugins/content_script/` to support the rich text editor.


# Testing

1. Build the `content_script` example plugin and add it as a development plugin to Joplin.
    - It may be necessary to upgrade its version of TypeScript and Webpack to do this.
2. Create a note with the following content:
    ````markdown
    Test
    ```justtesting
    foo
    bar
    baz
    ```
    ````
4. Switch to the Rich Text Editor.
5. Click on the generated link and check the console for a `Got response in content script: justtesting+response` message.
6. Make a change to the text just above the rendered block.
7. Switch back to the markdown editor
8. Verify that the original markup for the `justtesting` block is still present.
